shared/chat/blocking/block-modal.tsx

```diff
@@ -39,23 +43,113 @@ const CheckboxRow = (props: CheckboxRowProps) => {
   )
 }
 const reasons = ["I don't know this person", 'Spam', 'Harassment', 'Obscene material', 'Other...']
-const BlockModal = (props: Props) => {
-  const onFinish = todo
+const BlockModal = (props: OwnProps) => {
```

Comment:
I think this is too complex. Could it be factored out into smaller pieces? And/or be turned into a class.